StochasticBarrierFunctions/benchmarks/barrier_synthesis.jl

```diff
@@ -0,0 +1,286 @@
+                                                                    distribution_guided = distribution_guided), 
+                                       probabilities, initial_region, obstacle_region; time_horizon = time_horizon)
+    return res_pwc
+end
+
+function pwc_optimization_call(config, probabilities, initial_region, obstacle_region, barrier_type::GD_ALG)
+    # Optimize: method 4 (project gradient descent approach)
+    num_iterations, initial_lr, decay, momentum, time_horizon = get_kwargs(config, barrier_type::GD_ALG)
+    @time res_pwc = synthesize_barrier(GradientDescentAlgorithm(num_iterations = num_iterations, initial_lr = initial_lr,
+                                                                decay = decay, momentum = momentum), 
+                                       probabilities, initial_region, obstacle_region; time_horizon = time_horizon)
+    return res_pwc
+end
+
+function get_kwargs(config, barrier_type::SOS)
+    barrier_degree = get(config["barrier_settings"], "barrier_degree", SumOfSquaresAlgorithm().barrier_degree)
+    lagrange_degree = get(config["barrier_settings"], "lagrange_degree", SumOfSquaresAlgorithm().lagrange_degree) 
+    time_horizon = get(config["barrier_settings"], "time_horizon", 1) 
+    return barrier_degree, lagrange_degree, time_horizon
+end
+
+function get_kwargs(config, barrier_type::DUAL_ALG)
+    return get(config["barrier_settings"], "time_horizon", 1) 
+end
+
+function get_kwargs(config, barrier_type::CEGS_ALG)
+    δ = get(config["barrier_settings"], "iteration_δ", IterativeUpperBoundAlgorithm().δ)
+    num_iterations = get(config["barrier_settings"], "num_iterations", IterativeUpperBoundAlgorithm().num_iterations)
+    barrier_guided = get(config["barrier_settings"], "barrier_guided", IterativeUpperBoundAlgorithm().barrier_guided)
+    distribution_guided = get(config["barrier_settings"], "distribution_guided", IterativeUpperBoundAlgorithm().distribution_guided)
+    time_horizon = get(config["barrier_settings"], "time_horizon", 1) 
+    return δ, num_iterations, barrier_guided, distribution_guided, time_horizon
+end
+
+function get_kwargs(config, barrier_type::GD_ALG)
+    num_iterations = get(config["barrier_settings"], "num_iterations", GradientDescentAlgorithm().num_iterations)
+    initial_lr = get(config["barrier_settings"], "initial_lr", GradientDescentAlgorithm().initial_lr)
+    decay = get(config["barrier_settings"], "decay", GradientDescentAlgorithm().decay)
+    momentum = get(config["barrier_settings"], "momentum", GradientDescentAlgorithm().momentum)
+    time_horizon = get(config["barrier_settings"], "time_horizon", 1) 
+    return num_iterations, initial_lr, decay, momentum, time_horizon
+end
+
+function print_to_txt(system_flag, barrier_type, res)
+    # Print to txt                                   
+    file_path = "$(system_flag)/results/$(barrier_type)/result.txt"
+    open(file_path, "w") do file
+        println(file, res)
```

Comment:
```suggestion
        show(file, res)
```
I don't find the results files super readable, in part because there's no whitespace or indentation. By using `show(file, res)` I think it might include those (although I might be wrong).